### PR TITLE
SVC: Wake up the thread after the delay in WaitSync1

### PR DESCRIPTION
**Pull request by Subv**

### CODE REVIEW

File: src/core/hle/svc.cpp [hunk 1]
```diff
@@ -132,6 +129,8 @@ static Result WaitSynchronization1(Handle handle, s64 nano_seconds) {
 
     // Check for next thread to schedule
     if (wait.Succeeded() && *wait) {
+        // Create an event to wake the thread up after the specified nanosecond delay has passed
+        Kernel::WakeThreadAfterDelay(Kernel::GetCurrentThread(), nano_seconds);
```
Comment (yuriks) [inline src/core/hle/svc.cpp]: Does `WakeThreadAfterDelay` handle infinte timeouts?

Comment (Subv) [inline src/core/hle/svc.cpp]: Yep

Comment (yuriks) [inline src/core/hle/svc.cpp]: If you say so.

**Comment by yuriks** [comment]
LGTM

**Comment by purpasmart96** [comment]
@Subv Can you this for https://github.com/citra-emu/citra/pull/456/files#diff-4c2dcf02e3b93a7686c1362ff9f2ce5dR141 ?

**Comment by Subv** [comment]
@purpasmart96 WaitSynchronizationN has some outstanding issues which prevent it from working correctly (anything that uses it is completely broken), adding this to it would only make it worse as of now.

**Comment by purpasmart96** [comment]
@Subv Alright, then. LGTM

**Comment by bunnei** [comment]
LGTM

**Pull request merged**